### PR TITLE
Use text/template instead of html/template for binary url

### PR DESCRIPTION
**Pull request by soider**
Signed-off-by: Mikhail Sakhnov <msakhnov@mirantis.com>

## Description
The html/template claims to be template renderer engine which outputs html
It sanitizes outputs the same way as html by adding escaping

That breaks final URL if version or tool name or any other used field has different special signs in it

## Motivation and Context
If there are + signs or any other escaping signs in the version or name, the final url is broken.

## How Has This Been Tested?
By running existed test suite and manually downloading few tools

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [X] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [X] All new and existing tests passed.

- [ ] I have tested this on arm, or have added code to prevent deployment

### CODE REVIEW

**Comment by alexellis** [comment]
Hi @soider as part of our contributing process, can you raise an issue for this please and give a specific use-case that shows a template failing?

Thanks,

Alex

**Comment by alexellis** [comment]
In the meantime I'll approve the CI to run.

File: pkg/get/tools.go [hunk 1]
```diff
@@ -1310,5 +1313,69 @@ https://github.com/{{.Owner}}/{{.Repo}}/releases/download/{{.Version}}/{{.Name}}
...
+			{{- end -}}
+			{{ if HasPrefix .OS "ming" -}}
+			{{$ext = ".exe"}}
+			{{- end -}}
+			{{.Name}}-{{.Version}}-{{$arch}}{{$ext}}
+			`,
+		},
+	)
+	// k0s binary is available only for linux and windows
+	if runtime.GOOS != "darwin" {
```
Comment (alexellis) [inline pkg/get/tools.go:1339]: We have code to handle this, so would just remove that if guard.

**Review by alexellis — CHANGES_REQUESTED** [review]
This is shaping up very nicely. Could you just take a look at the note I left for you and amend that last commit and force push after?

**Comment by alexellis** [comment]
@yankeexe we have a PR for K0s already here, so we may want to close #454

**Comment by alexellis** [comment]
@soider please see the error: https://github.com/alexellis/arkade/pull/447/checks?check_run_id=3056341788

**Comment by alexellis** [comment]
FYI we have a rebase conflict on this now too. Could you fix? https://www.youtube.com/watch?v=8j0H6urZ-bU

**Comment by soider** [comment]
I resolved the conflict and dropped the check for darwin OS

**Comment by soider** [comment]
@alexellis as I mentioned, I dropped the OS check in the tools list, though I still able to download binary on darwin (while the binary doesn't support darwin)

**Comment by alexellis** [comment]
We have another rebase issue, sorry about that - due to merging #457 

Once resolved, I'll merge this and then make a recommendation about Darwin. We can detect the OS during the string building template, so that might be a good place to add extra text that would make it fail to match a valid binary.

**Comment by soider** [comment]
@alexellis rebased on master branch

**Comment by alexellis** [comment]
Merged, thank you for the patience.

**Comment by soider** [comment]
@alexellis and thank you for the hints as well!

**Comment by alexellis** [comment]
Want to try that suffix that's normally empty but breaks the URL delicately for Darwin?

**Comment by soider** [comment]
@alexellis I am not sure I understand what are you about. Could you explain a little further?

**Comment by alexellis** [comment]
Didn't you tell me that k0s downloads in error on Darwin?

Break its URL on purpose so that it gives the error saying that it's unavailable.

**Comment by soider** [comment]
Oh, yeah, got you. Good point actually, if the k0s changed the url scheme, I'd re-iterate the changes introduced here.

**Pull request merged**